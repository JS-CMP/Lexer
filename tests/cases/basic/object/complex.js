var person = {
    firstName: "John",
    lastName: "Doe",
    age: 30,
    address: {
        street: "123 Main St",
        city: "Anytown",
        state: "CA",
        zipCode: "12345"
    },
    contactInfo: {
        email: "john.doe@example.com",
        phone: "555-123-4567"
    },
    isEmployed: true,
    job: {
        title: "Software Engineer",
        company: "TechCorp",
        location: {
            city: "Silicon Valley",
            state: "CA"
        }
    },
    socialMedia: {
        twitter: "@johndoe",
        linkedin: "linkedin.com/in/johndoe"
    }
};