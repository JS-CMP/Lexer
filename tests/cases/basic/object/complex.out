#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {JS::Any person;person=JS::Any(std::make_shared<JS::Object>(std::unordered_map<std::u16string, JS::Any>{{u"firstName", JS::Any("John")},{u"lastName", JS::Any("Doe")},{u"age", JS::Any(30)},{u"address", JS::Any(std::make_shared<JS::Object>(std::unordered_map<std::u16string, JS::Any>{{u"street", JS::Any("123 Main St")},{u"city", JS::Any("Anytown")},{u"state", JS::Any("CA")},{u"zipCode", JS::Any("12345")}}))},{u"contactInfo", JS::Any(std::make_shared<JS::Object>(std::unordered_map<std::u16string, JS::Any>{{u"email", JS::Any("john.doe@example.com")},{u"phone", JS::Any("555-123-4567")}}))},{u"isEmployed", JS::Any(true)},{u"job", JS::Any(std::make_shared<JS::Object>(std::unordered_map<std::u16string, JS::Any>{{u"title", JS::Any("Software Engineer")},{u"company", JS::Any("TechCorp")},{u"location", JS::Any(std::make_shared<JS::Object>(std::unordered_map<std::u16string, JS::Any>{{u"city", JS::Any("Silicon Valley")},{u"state", JS::Any("CA")}}))}}))},{u"socialMedia", JS::Any(std::make_shared<JS::Object>(std::unordered_map<std::u16string, JS::Any>{{u"twitter", JS::Any("@johndoe")},{u"linkedin", JS::Any("linkedin.com/in/johndoe")}}))}}));}
